### PR TITLE
DF-760 auth token returned in a header

### PR DESCRIPTION
**Pull request by dvoracek-martin**

### CODE REVIEW

File: web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/JobAdvertisementApiRestController.java [hunk 1]
```diff
@@ -40,11 +51,12 @@ public JobAdvertisementApiRestController(JobAdvertisementApplicationService jobA
...
     @PostMapping
-    @ResponseStatus(HttpStatus.CREATED)
-    public ApiJobAdvertisementDto createFromApi(@RequestBody @Valid ApiCreateJobAdvertisementDto apiCreateJobAdvertisementDto) throws AggregateNotFoundException {
+    public ResponseEntity<ApiJobAdvertisementDto> createFromApi(@RequestBody @Valid ApiCreateJobAdvertisementDto apiCreateJobAdvertisementDto) throws AggregateNotFoundException {
         CreateJobAdvertisementDto createJobAdvertisementDto = jobAdvertisementFromApiAssembler.convert(apiCreateJobAdvertisementDto);
-        JobAdvertisementId jobAdvertisementId = jobAdvertisementApplicationService.createFromApi(createJobAdvertisementDto);
-        return jobAdvertisementToApiAssembler.convert(jobAdvertisementApplicationService.getById(jobAdvertisementId));
+        JobAdvertisement jobAdvertisement = jobAdvertisementApplicationService.createFromApi(createJobAdvertisementDto);
+        HttpHeaders headers = new HttpHeaders();
+        headers.add("owner_access_token", jobAdvertisement.getOwner().getAccessToken());
```
Comment (cjpsycos) [inline web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/JobAdvertisementApiRestController.java]: call it only "token"

File: application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementApplicationService.java [hunk 2]
```diff
@@ -144,14 +144,14 @@ public JobAdvertisementId createFromWebForm(CreateJobAdvertisementDto createJobA
 		return jobAdvertisement.getId();
 	}
 
-	public JobAdvertisementId createFromApi(CreateJobAdvertisementDto createJobAdvertisementDto) {
+	public JobAdvertisement createFromApi(CreateJobAdvertisementDto createJobAdvertisementDto) {
```
Comment (cjpsycos) [inline application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementApplicationService.java]: you should not give out the domain object out of the service.

you could create a dto with id and token.

Comment (dvoracek-martin) [reply application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementApplicationService.java]: all done

File: web/src/test/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/JobAdvertisementApiRestControllerIntTest.java [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        apiCreateJobAdvertisementDto.setApplyChannel(new ApiApplyChannelDto());
+
+        ResultActions post = post(apiCreateJobAdvertisementDto, URL);
+        post.andExpect(status().isCreated());
+
+        String contentAsString = post.andReturn().getResponse().getContentAsString();
+        JSONArray ja = new JSONArray("[" + contentAsString + "]");
+
+        // then
+        // TODO
```
Comment (cjpsycos) [inline web/src/test/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/JobAdvertisementApiRestControllerIntTest.java]: Please implement TODO

File: web/src/test/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/JobAdvertisementApiRestControllerIntTest.java [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    }
+
+    @Test
+    @WithApiUser
+    public void testCreateFavouriteItem() throws Exception {
+        // given
+        this.index(createJob(job01.id()));
+
+        //when
+        ApiCreateJobAdvertisementDto apiCreateJobAdvertisementDto = new ApiCreateJobAdvertisementDto();
```
Comment (fabiangotzen) [inline web/src/test/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/JobAdvertisementApiRestControllerIntTest.java]: Could be extracted into a ApiCreateJobAdvertisementFixture with a static factory method

Comment (dvoracek-martin) [reply web/src/test/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/JobAdvertisementApiRestControllerIntTest.java]: done

**Pull request merged**